src/Compilers/CSharp/Portable/Symbols/Source/ExplicitInterfaceHelpers.cs

```diff
@@ -17,21 +17,90 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
+#nullable enable
+    internal sealed class ExplicitInterfaceMemberInfo
+    {
+        private readonly SyntaxReference _explicitInterfaceSpecifierReference;
+        private readonly string _name;
```

Comment:
Do we really need to store these in the object? It feels like a caller should be able to get them from the declaration and supply them to the `Bind` method. Perhaps the bind method should be static and should be responsible for creating the object with the result values right when those values are actually needed.